renderdoc/driver/gl/gl_replay.cpp

```diff
@@ -2236,103 +2236,117 @@ byte *GLReplay::GetTextureData(ResourceId tex, uint32_t arrayIdx, uint32_t mip,
     arraysize = texDetails.depth;
   }
 
-  if(params.remap && intFormat != eGL_RGBA8 && intFormat != eGL_SRGB8_ALPHA8)
+  if(params.remap)
   {
-    RDCASSERT(params.remap == eRemap_RGBA8);
+    GLenum remapFormat = eGL_RGBA8;
+    if(params.remap == eRemap_RGBA8)
+      remapFormat = eGL_RGBA8;
+    else if(params.remap == eRemap_RGBA16)
+      remapFormat = eGL_RGBA16;
+    else if(params.remap == eRemap_RGBA32)
+    {
+      if(!IsGLES || HasExt[EXT_color_buffer_float])
+        remapFormat = eGL_RGBA32F;
+      else
+        remapFormat = eGL_RGBA32UI;
```

Comment:
I mean that the code is currently written under the assumption that it's always writing to a floating point target, whether it be 8-bit preview or 32-bit float for value picking or texture saving etc. I don't want to start requiring that it also work for writing to uint, at least not without a good reason.

24-bit depth isn't an integer format - it's unsigned normalised (unorm), so in the remapping it should be remapped to 0.0 - 1.0 in the float32 target.